packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -777,14 +775,36 @@ void main() {
           ];
         },
       );
+    }
 
+    // Popup a menu without any offset.
     await tester.pumpWidget(
       MaterialApp(
         home: Scaffold(
-          body: Center(
-            child: Material(
-              child: popupMenuButton,
-            ),
+          body: Material(
+            child: buildMenuButton(),
+          ),
+        ),
+      ),
+    );
+
+    // Popup the menu.
+    await tester.tap(find.byType(IconButton));
+    await tester.pumpAndSettle();
+
+    // Initial state, the menu start at Offset(8.0, 8.0), the 8 pixels is edge padding when offset.dx < 8.0.
+    expect(tester.getTopLeft(find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_PopupMenu<int?>')), const Offset(8.0, 8.0));
+
+    // Collapse the menu.
+    await tester.tap(find.byType(IconButton));
+    await tester.pumpAndSettle();
+
+    // Popup a new menu with Offset(50.0, 50.0).
```

Comment:
If the `offset` exceeds the midline of the screen, the actual menu offset will be reversed, more information refer to https://github.com/flutter/flutter/blob/55289324c6d1db9071fb372f9f15f5ad5a064eeb/packages/flutter/lib/src/material/popup_menu.dart#L624

Reply:
Thanks for the reference!